research-software-directory/RSD-as-a-service
# 1214 community settings

# Community module hidden by default

Closes #1214 

Changes proposed in this pull request:
* Added `modules` property to settings.json. It enables to define main menu items shown in the header of the RSD
* By default modules includes software, projects and organisation values. Communities is not shown
* If modules prop is not present in the settings.json all menuItems values are shown (in this case communities will be shown)
* Documentation of settings is extended
* Documentation of rsd admin section is extended with communities  
* Documentation dependcies are upgraded (docusaurus) and "locked" to exact version

How to test:
* `make start` to build and generate test data
* confirm that Communities section is not shown in the RSD menu
* add `communities` to /frontend/public/settings.json, "modules" and run `make start` again
```json
...
"host": {
   ...
    "modules":["software","projects","organisations","communities"]
  },
...
```
* confirm that communities section is shown
* remove some of the other modules and rebuild and confirm that "feature" works

PR Checklist:
